static/js/Promotions.jsx

```diff
@@ -1,200 +1,217 @@
-    function createBannerHtml(matchingAd) {
-        return `<div id="bannerTextBox">
-	<span class="${context.interfaceLang === "hebrew" ? "int-he" : "int-en" }" style="font-weight: bold">
-        ${matchingAd.bodyText}
-    </span>
-</div>
-<div id="bannerButtonBox">
-	<a class="button white ${context.interfaceLang === "hebrew" ? "int-he" : "int-en" }" 
-	    href="${matchingAd.buttonUrl}"   
-	    onclick='gtag("event", "promo_clicked", {  
-            campaignID: matchingAd.campaignId, adType:matchingAd.adType
-            })' 
-        target="_blank">
-        <span>${matchingAd.buttonText}</span>
-    </a>
-</div>`
-    }
-
-    function styleAds() {
-        if (adType === "banner") {
-            const matchingAd = matchingAds[0] // Only allow a single banner
-            if (!matchingAd) {return null}
-            const bannerHtml = createBannerHtml(matchingAd);
-            return <InterruptingMessage
-            messageName={matchingAd.campaignId}
-            messageHTML={bannerHtml}
-            style="banner"
-            repetition={matchingAd.repetition}
-            onClose={rerender} />
-        } else {
-              const sidebarAds = matchingAds.map(ad => <SidebarAd matchingAd={ad} key={ad.campaignId} />);
-              return (sidebarAds)
-        }
-    }
+  return matchingAds
+    ? matchingAds.map((ad) => (
+        <SidebarAd context={context} matchingAd={ad} key={ad.campaignId} />
+      ))
+    : null;
+};
 
-    return matchingAds ? styleAds() : null
+function trackSidebarAdImpression(ad) {
+  console.log(ad.campaignId + " has been seen");
+  gtag("event", "promo_viewed", {
+    campaignID: ad.campaignId,
+    adType: "sidebar",
+  });
+}
 
+function trackSidebarAdClick(ad) {
+  gtag("event", "promo_clicked", {
+    campaignID: ad.campaignId,
+    adType: "sidebar",
+  });
 }
 
-const SidebarAd = ({matchingAd}) => {
-    const classes = classNames({
-        sidebarPromo: 1,
-        blue: matchingAd.hasBlueBackground,
-    })
+const SidebarAd = React.memo(({ context, matchingAd }) => {
+  const classes = classNames({
+    sidebarPromo: 1,
+    blue: matchingAd.hasBlueBackground,
+  });
 
-    function getButton() {
-        return <a className={matchingAd.buttonStyle} href={matchingAd.buttonUrl}
-                  onClick={() => gtag("event", "promo_clicked", {
-                              campaignID: matchingAd.campaignId,
-                              adType:matchingAd.adType
-                            })}>
-        <img src={`/static/icons/${matchingAd.buttonIcon}`} aria-hidden="true" />
-        {matchingAd.buttonText}</a>
-    }
+  function getButton() {
+    return (
+      <a
+        className="button small"
+        href={matchingAd.buttonURL}
+        onClick={() => trackSidebarAdClick(matchingAd)}
+      >
+        {matchingAd.buttonIcon?.data ? (
+          <img
+            src={STRAPI_INSTANCE + matchingAd.buttonIcon.data.attributes.url}
```

Comment:
Just did some tests for this. It doesn't affect the number of API calls made, but I'm pretty sure it involves bandwidth usage. I can't know for sure though on the bandwidth usage, because the counter on the dashboard hasn't updated when I conducted a test.